pages/your_tasks/virtual_dev_environments.md

```diff
@@ -0,0 +1,120 @@
+without affecting the working version of your project. You need to set up a separate virtual environment for your branch to
+'isolate' your code while testing the new feature.
+- You do not have to worry too much about specific versions of external libraries that your project depends on most of the time.
+Virtual environments enable you to always use the latest available version without specifying it explicitly.
+They also enable you to use a specific older version of a package for your project, should you need to. 
+
+### Solutions
+
+- Make your research software reusable and your research that relies on that software reproducible by setting up and sharing its virtual development environment.
+
+## How do you create virtual software development environments? 
+
+### Description
+
+Most modern programming languages use some kind of virtual environments or a similar mechanism to isolate libraries or dependencies for a specific project, 
+making it easier to develop, run, test and share code with others. 
+Some examples include Bundler for Ruby, Conan for C++, or Maven with classpath for Java. 
+This can also be achieved with more generic package (dependency) managers like Spack, which is used extensively in HPC settings to resolve complex dependencies. 
+
+Part of managing a virtual software development environment involves installing, updating and removing external packages on your system. 
+You would need a package manager tool for your programming language to be able to do that - this is typically a command line tool that you invoke from 
+a command line terminal. 
+In addition to a package manager, you will need another command line tool to create and manage virtual environments on your machine. 
+Sometimes, a package manager combines both of these functionalities and you only need to install one extra tool on your system.
+
+### Considerations
+
+- There are often multiple package and environment management tools for a single programming language. For example,
+there are several commonly used command line tools for managing Python packages and virtual environments to choose from:
+  - `pip` - Python package manager tool which interacts and obtains the packages from the central repository called Python Package Index (PyPI). `pip` can now be used with all Python distributions (including Anaconda).
+  - `venv` - Python virtual environment manager tool available by default from the standard Python distribution from Python 3.3+
+  - `venv` - Python virtual environment manager tool available by default from the standard Python distribution from Python 3.3+
+  - `virtualenv` - Python virtual environment manager tool which needs to be installed separately (does not come with Python distribution) but supports both Python 2.7+ and Python 3.3+ versions
+  - `pipenv` - Python package and environment management sysyem created to fix certain shortcomings of `virtualenv`
```

Comment:
might be worth adding a note that pipenv and pip-venv are not the same thing (and maybe a line on how they are different)